### DISCUSSION TITLE
How to keep the user logged in?

### DISCUSSION THREAD

**Discussion by gomflo** [supabase/supabase#568] [Questions]
Im using the following example: [https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth](https://github.com/supabase/supabase/tree/master/examples/nextjs-with-supabase-auth), but the session only lasts for about 30 minutes? i want the session to last for at least the duration of the cookie.

Thanks!

**Comment by Vinzent03** [comment]
I agree with you. I would love some answer from mods. I'm using Supabase in flutter and already use `auth.recoverSession()`, but it's invalid after an hour or so.

**Comment by ChronSyn** [comment]
I believe this can be changed from the dashboard.

Go to Authentication > Settings and change JWT expiry:
![image](https://user-images.githubusercontent.com/12896049/125219218-5c961200-e2bc-11eb-9a61-502e32347fc4.png)

**Reply by Vinzent03** [reply]
Thanks for the reply, but the maximum is one week. Is there no way to keep the user logged in forever, like with Firebase Auth or any other mobile app?

**Reply by ChronSyn** [reply]
This is just the JWT duration, not the maximum length they can be logged in from their perspective.


The way I understand it, is that as long as the user has been active within that time, their auth token should have been exchanged for a new one (using a refresh token mechanism) and they are essentially given a new token that can last another week (if your JWT expiry time is set that high). If they aren't active in that time, their tokens will likely be expired.

With that said, I haven't had any problems with my own projects, even with a modest JWT expiry of 3000. I'd check to make sure that the cookies aren't being removed by a browser extension or similar.

**Reply by Vinzent03** [reply]
Okay, but if the user is more than a week offline, he has to log in again, right? That's still uncommon for mobile apps. (sorry replied in the wrong thread before)
